nemo/collections/nlp/models/text_classification/text_classification_model.py

```diff
@@ -159,31 +204,36 @@ def _setup_tokenizer(self, cfg: DictConfig):
         self.tokenizer = tokenizer
 
     def setup_training_data(self, train_data_config: Optional[DictConfig]):
-        self._train_dl = self._setup_dataloader_from_config(cfg=train_data_config)
+        self._train_dl = self._create_dataloader_from_config(cfg=train_data_config, mode='train')
 
     def setup_validation_data(self, val_data_config: Optional[DictConfig]):
-        self._validation_dl = self._setup_dataloader_from_config(cfg=val_data_config)
+        self._validation_dl = self._create_dataloader_from_config(cfg=val_data_config, mode='validation')
 
     def setup_test_data(self, test_data_config: Optional[DictConfig]):
-        self._test_dl = self._setup_dataloader_from_config(cfg=test_data_config)
+        self._test_dl = self._create_dataloader_from_config(cfg=test_data_config, mode='test')
 
-    def _setup_dataloader_from_config(self, cfg: DictConfig):
-        input_file = cfg.file_name  # os.path.join(self.dataset_cfg.data_dir, f"{cfg.file_name}")
+    def _create_dataloader_from_config(self, cfg: DictConfig, mode):
+        if not cfg or not cfg.file_path:
+            logging.info(
+                f"Dataloader config or file_path for the {mode} is missing, so no data loader for test is created!"
+            )
+            return None
+        input_file = cfg.file_path
         if not os.path.exists(input_file):
             raise FileNotFoundError(
                 f'{input_file} not found! The data should be be stored in TAB-separated files \n\
-                "validation_ds.file_name" and "train_ds.file_name" for train and evaluation respectively. \n\
+                "validation_ds.file_path" and "train_ds.file_path" for train and evaluation respectively. \n\
                 Each line of the files contains text sequences, where words are separated with spaces. \n\
                 The label of the example is separated with TAB at the end of each line. \n\
                 Each line of the files should follow the format: \n\
                 [WORD][SPACE][WORD][SPACE][WORD][...][TAB][LABEL]'
             )
 
         dataset = TextClassificationDataset(
-            input_file=input_file,
             tokenizer=self.tokenizer,
+            input_file=input_file,
```

Comment:
@tkornuta-nvidia we can readlines() in the file and pass a list of str here. That way it would be the same arg, will this solve the new arg in the dataset issue you mentioned in the dataset class?

Reply:
@tkornuta-nvidia I thought it would make it easier for your efforts but if you want I can create another datasetclass just for inference.